packages/solana-wallet-adapter/src/SolanaWalletAdapterProvider.tsx

```diff
@@ -34,11 +34,7 @@ export default function SolanaWalletAdapterProvider({
     never
   >();
 
-  const connection = useMemo(() => {
-    return selectedChain?.rpcEndpoint
-      ? new Connection(selectedChain?.rpcEndpoint)
-      : undefined;
-  }, [selectedChain?.rpcEndpoint]);
+  const { connection } = useConnection();
```

Comment:
The selected chain uses the connection object to determine which chain should be selected.

```
useEffect(() => {
  if (wallet?.adapter.publicKey) {
    const chain = chains
      .filter((c) => c.type === SupportedChains.Solana)
      .filter((c) => c.rpcEndpoint === connection?.rpcEndpoint);

    if (selectedChain?.name !== chain[0]?.name) {
      setSelectedChain(chain[0]);
      return;
    }
  }
}, [
  chains,
  connection?.rpcEndpoint,
  setSelectedChain,
  wallet?.adapter.publicKey?.toBase58(),
  connected,
  publicKey?.toBase58(),
]);
```